[ADD] sale_order_shipping_date: added compute field shipping_date

suniagajose opened the pull request:
### [VX#6085 CA#9 HU#805](https://www.vauxoo.com/web#id=805&view_type=form&model=user.story)
## Summary

This module add a field into Sale Order that show the Date by which the are sure to be delivered. In constrast to commitment date, not only is taken into account the product lead time, but also the resupply delay, the also the resupply delay, the delivery lead time from supplier and the manufacturing delay.

Technically, the shipping date field comes to replace the commitment date.
## Test Case

**Sketch**

```
        e.g.
        - products have 1 day delay
        - sale was created on 2016-10-25 20:16:11
        -------------------------------
        product Apple Wireless Keyboard
        warehouse Chicago Warehouse
        date_expected False
        resupply_date 2016-10-28 20:16:11 (day when each one is available)
        manuacturing_date False
        purchase_date 2016-10-29 20:16:11
        -------------------------------
        product Blank CD
        warehouse Chicago Warehouse
        date_expected False
        resupply_date False
        manuacturing_date False
        purchase_date 2016-10-27 20:16:11
        -------------------------------
        product USB Adapter
        warehouse Chicago Warehouse
        date_expected 2016-10-25 20:16:11
        resupply_date 2016-10-28 20:16:11
        manuacturing_date False
        purchase_date False
        -------------------------------

        result:
        - resupply_date 2016-10-28 20:16:11 + 1 day (product delay)
        - shipping_date is 2016-10-29 20:16:11 (4 days after sale)
```

**Configure to compute resupply date**

Check _Default Resupply Warehouse_ in Chicago Warehouse
![pantallazo-2016-10-25 18-44-37](https://cloud.githubusercontent.com/assets/5335402/19708809/f81a75f4-9ae6-11e6-9afa-76414f48247a.png)

Go to Routes, and select _Chicago Warehouse: Supply Product from YourCompany_
![pantallazo-2016-10-25 18-44-38](https://docs.google.com/uc?id=0B5vKhatgX1y8d193anAwTU45TUU)

Select Pull Rule _Chic: Inter Company Transit -> Stock_
![pantallazo-2016-10-25 18-44-39](https://docs.google.com/uc?id=0B5vKhatgX1y8MUpMTFZtSFNfVlk)

Set a delay: the number of days that product will be in transit. 3 days for this example
![pantallazo-2016-10-25 18-44-40](https://docs.google.com/uc?id=0B5vKhatgX1y8Yzl6Nmp2R2VfVE0)

**Configure to compute purchase date**

Select Product _Apple Wireless Keyboard_ and check _Delivery Lead Time_ (4 days)
![pantallazo-2016-10-25 18-44-41](https://docs.google.com/uc?id=0B5vKhatgX1y8bTF4UVZNOUxtTW8)

**Configure to compute manufacturing date**

Ensure that _Customer Lead Time_ and Manufacturing Lead Time are 1 day for products: _Apple Wireless Keyboard_,  _Blank CD_, _USB Adapter_
![pantallazo-2016-10-25 18-44-42](https://docs.google.com/uc?id=0B5vKhatgX1y8S0xwOVBqcmFFWU0)

**Create Order for Test Case**

Create a Order with products: _Apple Wireless Keyboard_,  _Blank CD_, _USB Adapter_ at _Chicago Warehouse_ 
![pantallazo-2016-10-25 18-44-43](https://docs.google.com/uc?id=0B5vKhatgX1y8b0pxcDBqNlNTUW8)

Go to _Other Information_ tab and check the _Wholeness Date_, should be 4 days after _Date Order_ (check date order on previous image)
![pantallazo-2016-10-25 18-44-43](https://docs.google.com/uc?id=0B5vKhatgX1y8aTFCSjdZQlZUYlE)

**Conclusion**
- All products have +1 day delay
- Apple Wireless Keyboard is the last product to be receipt 
- The best choice for Apple Wireless Keyboard is resupply
- Resupply date for Apple Wireless Keyboard is 2016-10-28 20:16:11 (3 days after sale)
- To the resupply date is added +1 for the product delay, in order to compute de shipping date
- **Shipping date is 2016-10-29 20:16:11 (4 days after sale)**

suniagajose commented:
Hi @JulioSerna 

Can you do a functional review to this??

Thanks

hugho-ad commented on sale_order_wholeness_date/models/procurement.py:
```diff
@@ -0,0 +1,34 @@
+        # resupply rules. To optimize this date should be checked if the
+        # orderpoint would be satisfied by a buy order, manufacturing order or
+        # a resupply request, avoiding take the buy order as the only rule,
+        # which is how comes from Odoo.
```
This info could be between `""" Docstring"""`` marks

suniagajose replied:
you right!

thanks @hugho-ad

Commit:
[ADD] sale_order_shipping_date: added compute field shipping_date

This module add a field into Sale Order that show the Date
by which the are sure to be delivered. In contrast to commitment
date, not only is taken into account the product lead time, but
also the resupply delay, the also the resupply delay, the
delivery lead time from supplier and the manufacturing delay.

Technically, the shipping date field comes to replace the
commitment date.

Commit:
[IMP] workalendar_holidays: added logistic calendar to company model

The logistic calendar is used to load the work time, and load company
holidays, in order to be able to compute working days in company daily
operations.

suniagajose commented:
moved to https://github.com/Vauxoo/yoytec/pull/2141
